### PR TITLE
Add facebook login

### PR DESCRIPTION
**Pull request by tberkane**
This PR adds the possibility to sign in through facebook (only through the profile for now).

### CODE REVIEW

File: app/src/androidTest/java/com/github/epfl/meili/home/FacebookAuthenticationServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        fauth = FacebookAuthenticationService()
+    }
+
+    @Test
+    fun testA() {
+        assert(fauth.getCurrentUser() == null)
+    }
+
+    @Test
+    fun testB() {
```
Comment (MarcelTorne) [inline app/src/androidTest/java/com/github/epfl/meili/home/FacebookAuthenticationServiceTest.kt:32]: Nitpick: could you change the names of the tests to something more descriptive of what is being tested?

**Review by MarcelTorne — COMMENTED** [review]
Thanks for the work! Everything looks really good, just left 2 small nitpicks. I have a general question, I believe you are leaving the creation of the fragment for all logins for next sprint?

**Comment by tberkane** [comment]
> Thanks for the work! Everything looks really good, just left 2 small nitpicks. I have a general question, I believe you are leaving the creation of the fragment for all logins for next sprint?

Yes exactly

**Review by MarcelTorne — APPROVED** [review]
Thanks again for the work!

File: app/src/main/res/layout/activity_profile.xml [hunk 2]
```diff
@@ -295,6 +295,14 @@
         android:onClick="onProfileButtonClick"
         android:text="@string/sign_in" />
 
+
+    <com.facebook.login.widget.LoginButton
+        android:id="@+id/facebook_sign_in"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="50dp"
```
Comment (Bwananah) [inline app/src/main/res/layout/activity_profile.xml:303]: The button overlaps with the dark/light mode button. Just a nitpick

**Review by Bwananah — APPROVED** [review]
Just one little nitpick on the UI, but otherwise looks good. No need to change it if you prefer it where it is.

**Pull request merged**